app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -336,13 +336,10 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         // Launch this on a background thread so as not to affect startup performance
         lifecycleScope.launch(IO) {
             if (
-                settings().isDefaultBrowser() &&
-                settings().wasDefaultBrowserOnLastResume != settings().isDefaultBrowser()
+                settings().checkDefaultBrowserAndSet()
```

Comment:
**thought:** to be explicit about my understanding, the default browser is now resolved later during start up (dispatched from `HomeActivityonResume`) on the IO thread rather than synchronously on the main thread when settings is first accessed. This seems okay to me: the default browser should not change during the start up.

We could consider moving this to the visual completeness queue but it would functionally change the timing and probably not worth the effort: this is a small operation on a background thread so it's probably okay for now.

Reply:
I agree with this. I think I'd leave it this as is for now. However, moving this to visual completeness could help our start up still (possibly)